src/imgui/ImGuiDisassembly.cc

```diff
@@ -179,18 +179,25 @@ static void toggleBp(uint16_t addr, const BpLine& bpLine, std::span<BreakPoint>
 		addBp->setCondition(toTclExpression(getCurrentSlot(cpuInterface, debugger, addr)));
 	}
 }
+
+static std::optional<uint16_t> selectedAddr = {};
+
 void ImGuiDisassembly::actionToggleBp(MSXMotherBoard& motherBoard)
 {
 	auto pc = motherBoard.getCPU().getRegisters().getPC();
 	auto& cpuInterface = motherBoard.getCPUInterface();
 	auto& debugger = motherBoard.getDebugger();
 	auto& bps = cpuInterface.getBreakPoints();
-
-	auto bpLine = examineBpLine(pc, bps, cpuInterface, debugger);
-
 	std::optional<BreakPoint> addBp;
 	std::optional<unsigned> removeBpId;
-	toggleBp(pc, bpLine, bps, cpuInterface, debugger, addBp, removeBpId);
+
+	if (selectedAddr) {
+		auto bpLine = examineBpLine(*selectedAddr, bps, cpuInterface, debugger);
+		toggleBp(*selectedAddr, bpLine, bps, cpuInterface, debugger, addBp, removeBpId);
+	} else {
+		auto bpLine = examineBpLine(pc, bps, cpuInterface, debugger);
+		toggleBp(pc, bpLine, bps, cpuInterface, debugger, addBp, removeBpId);
+	}
```

Comment:
This can be simplified to:
```
auto addr = selectedAddr? *selectedAddr : motherBoard.getCPU().getRegisters().getPC();
```
And then mostly the original code, with `pc` replaced by `addr`.